tron/kubernetes.py

```diff
@@ -148,35 +148,58 @@ def handle_event(self, event: Event) -> None:
+                        # these Pods should be phase=Succeeded with an exit code of 0 - even though the container never actually
+                        # started. So far, we've noticed that when this happens, the finished_at and reason fields will be None
+                        # and thus we'll check for at least one of these conditions to detect an abnormal exit and actually "fail"
+                        # the affected action
+                        # NOTE: hopefully this won't change too drastically in future k8s upgrades without the actual problem (incorrect
+                        # success) being fixed :p
+                        if state_termination_metadata.get("exitCode") == 0 and (
+                            state_termination_metadata.get("finishedAt") is None
+                            and state_termination_metadata.get("reason") is None
+                        ):
+                            # -9 is EXIT_KUBERNETES_ABNORMAL from actionrun.py - we just can't import that value right now since there'd be a circular dependency
+                            exit_code = -9
+                            self.log.warning("Container never started due to a Kubernetes/infra flake!")
+                            self.log.warning(
+                                f"If automatic retries are not enabled, run `tronctl retry {self.id}` to retry."
+                            )
+                    elif k8s_type == "failed" or k8s_type == "killed":
+                        # Handling spot terminations
+                        if last_state_termination_metadata.get("exitCode", 1) == 137 and (
+                            "deleted" in last_state_termination_metadata.get("message", "")
+                            and last_state_termination_metadata.get("reason", "") == "ContainerStatusUnknown"
+                        ):
+                            exit_code = -10
+                            self.log.warning("Tronjob failed due to spot interruption.")
+                        # Handling K8s scaling down a node
+                        elif state_termination_metadata.get("exitCode", 1) == 143 and (
+                            state_termination_metadata.get("reason", "") == "Error"
+                        ):
+                            exit_code = -11
+                            self.log.warning("Tronjob failed due to Kubernetes scaling down a node.")
+                        if exit_code in [-10, -11]:
+                            self.log.warning(
+                                f"If automatic retries are not enabled, run `tronctl retry {self.id}` to retry."
+                            )
+                            self.log.warning(
+                                "If this job is idempotent, then please consider increasing the number of retries to your action. If your job is not idempotent then please set this job to run on the stable pool rather than the default."
```

Comment:
i'm a little worried about folks seeing this message and raising the retry count significantly when their currently configured # of retries is fine (e.g., they got unlucky once with a spot termination, but then the automatic retry worked)

should we reword this to `If this action is idempotent, then please consider enabling automatic retries for your action. If your action is not idempotent, then please configure this action to run on the stable pool rather than the default.`